packages/serverpod/lib/src/server/message_central.dart

```diff
@@ -168,4 +170,33 @@ class MessageCentral {
       }
     }
   }
+
+  /// Creates a stream that listens to a specified channel.
+  ///
+  /// This stream emits messages of type [T] whenever a message is received on
+  /// the specified channel.
+  ///
+  /// If messages on the channel does not match the type [T], the stream will
+  /// emit an error.
+  Stream<T> createStream<T>(
+    Session session,
+    String channelName,
+  ) {
+    var controller = StreamController<T>();
+    void addToStream(dynamic message) {
+      try {
+        controller.add(message as T);
+      } catch (e) {
+        controller.addError(e);
+      }
+    }
+
+    addListener(session, channelName, addToStream);
+
```

Comment:
To investigate: Need to validate that the stream is closed if the listener is removed from some other source.